.github/workflows/ci-testing-deploy.yml

```diff
@@ -2864,3 +2864,56 @@ jobs:
         env:
           TAG_PREFIX: hotfix-staging-github
         run: ./ci/deploy/dockerhub-deploy.bash -n
+
+  check-api-specs-updated:
+    needs: [changes]
+    if: ${{ needs.changes.outputs.anything-py == 'true' || github.event_name == 'push' }}
+    timeout-minutes: 10
+    name: check api-specs are up to date
+    runs-on: ubuntu-latest
+    steps:
+      - name: setup python environment
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+      - name: install uv
+        uses: astral-sh/setup-uv@v4
+        with:
+          version: "0.4.x"
+          enable-cache: false
+      - name: checkout source branch
+        uses: actions/checkout@v4
+      - name: Regenerate specs and check
+        run: |
```

Comment:
you might want to put this command and the one blow in a script inside the `ci/github/helpers`. This way if we need to run the command locally (say CI is failing) it's simpler.